src/components/autocomplete/js/highlightController.js

```diff
@@ -2,26 +2,33 @@ angular
     .module('material.components.autocomplete')
     .controller('MdHighlightCtrl', MdHighlightCtrl);
 
-function MdHighlightCtrl ($scope, $element, $interpolate) {
+function MdHighlightCtrl ($scope, $element, $attrs) {
   this.init = init;
 
-  return init();
+  function init (termExpr, unsafeTextExpr) {
+    var text = null,
+        regex = null,
+        flags = $attrs.mdHighlightFlags || '',
+        watcher = $scope.$watch(function($scope) {
+          return {
+            term: termExpr($scope),
+            unsafeText: unsafeTextExpr($scope)
```

Comment:
Previously, it depended on the fact that ng-repeat would destroy and create a new instance of the directive if the underlying data changed. Virtual repeat reuses dom nodes (and thus the directives they contain), so they need to update on scope change.